tt_metal/llrt/tt_cluster.hpp

```diff
@@ -264,28 +265,28 @@ class Cluster {
 
     // There is one device driver per PCIe card. This map points id of the MMIO device points to the associated device
     // driver
-    std::unordered_map<chip_id_t, std::unique_ptr<tt_device>> mmio_device_id_to_driver_;
+    tt::IndexMap<std::unique_ptr<tt_device>> mmio_device_id_to_driver_;
 
     // Need to hold reference to cluster descriptor to detect total number of devices available in cluster
     // UMD static APIs `detect_available_device_ids` and `detect_number_of_chips` only returns number of MMIO mapped
     // devices
     std::string cluster_desc_path_;
     std::unique_ptr<tt_ClusterDescriptor> cluster_desc_;
     // There is an entry for every device that can be targeted (MMIO and remote)
-    std::unordered_map<chip_id_t, metal_SocDescriptor> sdesc_per_chip_;
+    tt::IndexMap<metal_SocDescriptor> sdesc_per_chip_;
 
     // Collections of devices that are grouped based on the associated MMIO device. MMIO device is included in the
     // grouping
-    std::unordered_map<chip_id_t, std::set<chip_id_t>> devices_grouped_by_assoc_mmio_device_;
+    tt::IndexMap<std::set<chip_id_t>> devices_grouped_by_assoc_mmio_device_;
     // Save mapping of device id to associated MMIO device id for fast lookup
-    std::unordered_map<chip_id_t, chip_id_t> device_to_mmio_device_;
+    tt::IndexMap<chip_id_t> device_to_mmio_device_;
```

Comment:
same here, there is an entry for each device so we don't necessarily need the IndexMap here